internal/util/update/update_test.go

```diff
@@ -247,12 +247,12 @@ func TestCommand_Run_noCommit(t *testing.T) {
 				Pkg:      pkgtest.CreatePkgOrFail(t, g.LocalWorkspace.FullPackagePath()),
 				Strategy: strategy,
 			}.Run(fake.CtxWithDefaultPrinter())
-			if !assert.Error(t, err) {
-				return
-			}
-			assert.Contains(t, err.Error(), "contains uncommitted changes")
-
-			if !g.AssertLocalDataEquals(testutil.Dataset3, false) {
+			if strategy == kptfilev1.FastForward {
```

Comment:
Earlier test was testing "if there are uncommitted changes in the pkg", `kpt pkg update` will fail for all the update strategies. Not sure what is this modified test is testing ?
One way to think will be:
Modified test now becomes `kpt pkg update` tests. if there are other `pkg update` tests, then this will be duplication of those, so we should think of removing duplicate test.
If we see want to preserve these tests, then behavior for all the update strategies is not clear in the modified test.

Reply:
Agree. I was on the fence to remove these tests but if you feel the same it is better to remove these as they are duplicates. The case where update should work with uncommited changes is tested in cmdupdate_test.go above.

Reply:
ok. +1 for removing duplicate tests. We also need to keep our test matrix (the spread-sheet) up-to-date.

Reply:
Done. You can approve this PR if the rest looks good.